TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs

```diff
@@ -0,0 +1,137 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using SimpleJSON;
+
+public class TreasureSetupController : MonoBehaviour {
+
+	public static TreasureSetupController instance = null;
+
+	void Awake()
+	{
+		if (instance == null)
+
+			instance = this;
+
+		else if (instance != this)
+
+			Destroy(gameObject);
+
+		DontDestroyOnLoad (gameObject);
+
+	}
+
+	void Start(){
+		if (!gameObject.GetComponent<NetworkManager> ().enabled) {
+			GetGameTreasure ("gg");		
+		}
+	}
+
+	public void GetGameTreasure(string userName){
+
+		string str = "{\"flag\":3, \"usn\":\"" + userName + "\"}";
+		if (!gameObject.GetComponent<NetworkManager> ().enabled) {
+			TextAsset jsonData = Resources.Load<TextAsset> ("TestForTreasureSetup");
+			var strJsonData = jsonData.text;
+			Debug.Log (strJsonData);
+			ForEachGame (strJsonData);
+		} else {
+			NetworkManager.instance.SendData (str);
+		}
+
+	}
+
+	//infos included in json:
+	//	flag,
+	//	game_id, game_name, treasure_count, maker_id, status,
+	//	treasure_id, treasure_name, description, game_id, location, point, catchgame_cat, target_img_url
+	public GameObject game;
+	public GameObject treasure;
+	public static List<string> m_Data = new List<string>();
+
+	public GameObject ForEachGame(string data){
+
+		var jsonData = JSON.Parse (data);
+
+		var games = jsonData ["Games"];
+
+		GameObject gameTreasurePanel = new GameObject();
+		gameTreasurePanel.tag = "GameTreasurePanel";
+
+		for (int i = 0; i < games.Count; i++) {
+			//make new game panel
+			GameObject newGame = (GameObject) Instantiate (game, new Vector3(0,0,0), Quaternion.identity);
+			GameAttributes ga = newGame.GetComponent<GameAttributes> ();
+			// attach game attributes
+			ga.SetAsChildOf(gameTreasurePanel);
+			ga.SetAttributes(games[i]["game_id"], games[i]["game_name"], games[i]["treasure_count"].AsInt, games[i]["maker_id"], games[i]["status"].AsInt);
```

Comment:
There's a whole bunch of `games[i]` going on here. What about pulling this out into a single variable to simplify some of this?